packages/mobile/src/exchange/ExchangeHomeScreen.tsx

```diff
@@ -1,126 +1,247 @@
+    currentExchangeRate,
+  ])
+  const dollarsToLocal = useCallback(
+    (amount) => convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
+
+  const currentGoldRateInLocalCurrency = dollarsToLocal(goldToDollars(1))
+  let rateChangeInPercentage, rateWentUp
+  const exchangeHistory = useSelector(exchangeHistorySelector)
+  if (exchangeHistory.aggregatedExchangeRates.length) {
+    const oldestGoldRateInLocalCurrency = dollarsToLocal(
+      exchangeHistory.aggregatedExchangeRates[0].exchangeRate
+    )
+    if (oldestGoldRateInLocalCurrency) {
+      const rateChange = currentGoldRateInLocalCurrency?.minus(oldestGoldRateInLocalCurrency)
+      rateChangeInPercentage = currentGoldRateInLocalCurrency
+        ?.div(oldestGoldRateInLocalCurrency)
+        .minus(1)
+        .multipliedBy(100)
+      rateWentUp = rateChange?.gt(0)
+    }
   }
 
-  goToBuyGold = () => {
+  const hasGold = new BigNumber(goldBalance || 0).isGreaterThan(0)
+
+  useEffect(() => {
+    dispatch(fetchExchangeRate())
+  }, [])
+
+  const headerOpacity = useCallback(
+    () => ({
+      opacity: animatedValue.interpolate({
+        inputRange: [0, 100],
+        outputRange: [0, 1],
+        extrapolate: Animated.Extrapolate.CLAMP,
+      }),
+    }),
+    []
+  )
+
+  const goToBuyGold = useCallback(() => {
     CeloAnalytics.track(CustomEventNames.gold_buy_start)
-    this.props.navigation.navigate(Screens.ExchangeTradeScreen, {
+    navigation.navigate(Screens.ExchangeTradeScreen, {
       makerTokenDisplay: {
         makerToken: CURRENCY_ENUM.DOLLAR,
-        makerTokenBalance: this.props.dollarBalance || '0',
+        makerTokenBalance: dollarBalance || '0',
       },
     })
-  }
+  }, [dollarBalance])
 
-  goToBuyDollars = () => {
+  const goToBuyDollars = useCallback(() => {
     CeloAnalytics.track(CustomEventNames.gold_sell_start)
-    this.props.navigation.navigate(Screens.ExchangeTradeScreen, {
+    navigation.navigate(Screens.ExchangeTradeScreen, {
       makerTokenDisplay: {
         makerToken: CURRENCY_ENUM.GOLD,
-        makerTokenBalance: this.props.goldBalance || '0',
+        makerTokenBalance: goldBalance || '0',
       },
     })
-  }
+  }, [dollarBalance])
 
-  render() {
-    const { t, goldBalance } = this.props
-    const hasGold = new BigNumber(goldBalance || 0).isGreaterThan(0)
+  const onScroll = useCallback(
+    Animated.event(
+      [
+        {
+          nativeEvent: {
+            contentOffset: {
+              y: (y: Animated.Node<number>) =>
+                Animated.block([
+                  Animated.set(animatedValue, y),
+                  Animated.call([y], ([offsetY]) => {
+                    setIsScrolled(offsetY > 0)
```

Comment:
Again maybe this is fine for now, but this is gonna trigger `setIsScrolled` every 50 ms as the user scrolls.
Would be better to use an `Animated.Value` for this, and then pass it to `DrawerTopBar`.

I've got an old Nexus 5 here where the scroll perf could be impacted very much by this.
I can try without and without this to see the difference.